mewarpx/scripts_ancillary/tinyprofiler_plotting_example.py

```diff
@@ -0,0 +1,270 @@
+        data = json.load(f)
+
+        gf = ht.GraphFrame.from_literal(data)
+        gf.drop_index_levels()
+
+        split_name = str(name).split("/")
+        cells = split_name[1]
+        cores = split_name[2].replace("core", "")
+
+        # sort by excl_max_percent
+        gf.dataframe = gf.dataframe.sort_values(by=parameter, ascending=False)
+        # filter to only high, exclusive percents
+        filtered_gf = gf.filter(lambda x : not math.isnan(x[parameter]) and x[parameter] > 10)
+        # Add an 'info' index, which will be the pivot point later
+        filtered_gf.dataframe["info"] = f"{cells}x{cells}\n{cores}"
+
+        if "gpu" not in cores:
+            filtered_gf.dataframe["info"] += "procs"
+
+        # add the dataframe to the list
+        dataframes.append(filtered_gf.dataframe)
+
+    # concat all the dataframes together
+    result = pd.concat(dataframes)
+    # pivot the dataframes around the 'info' index
+    pivot_df = result.pivot(index="info", columns="name", values=parameter)
+
+    # plot the data
+    fig = pivot_df.loc[:,:].plot.bar(stacked=True, figsize=(11,12),
+        title=description + " max percents > 10%",
+        ylabel=description + " max percent", xlabel="Simulation Parameters",
+    )
+
+    # this puts the legend outside of the plot
+    fig.legend(bbox_to_anchor=(1,1))
+
+    plt.savefig(description.lower() + "_max_percent_time.png", bbox_inches="tight")
+
+def plot_union_of_times(top_n, exclusive, avg):
+    function_names = set([])
+    parameter = get_parameter_name(exclusive, avg)
+    description = get_description(exclusive, avg)
+
+    for name in Path("warpx_outputs/").rglob("*.json"):
+        f = open(name)
+        data = json.load(f)
+        print("adding function names:", name)
+
+        gf = ht.GraphFrame.from_literal(data)
+        gf.drop_index_levels()
+
+        top = gf.dataframe.sort_values(by=parameter, ascending=False)[:top_n]
+        for function in top["name"]:
+            function_names.add(function)
+
+    dataframes = []
+    for name in Path("warpx_outputs/").rglob("*.json"):
+        print("plotting:", name)
+
+        f = open(name)
+        data = json.load(f)
+
+        gf = ht.GraphFrame.from_literal(data)
+        gf.drop_index_levels()
+
+        split_name = str(name).split("/")
+        cells = split_name[1]
+        cores = split_name[2].replace("core", "")
+
+        gf.dataframe = gf.dataframe.sort_values(by=parameter, ascending=False)
+
+        filtered_gf = gf.filter(lambda x : not math.isnan(x[parameter]) and x["name"] in function_names)
+
+        filtered_gf.dataframe["info"] = f"{cells}x{cells}\n{cores}"
+        if "gpu" not in cores:
+            filtered_gf.dataframe["info"] += "procs"
+
+        dataframes.append(filtered_gf.dataframe)
+
+    result = pd.concat(dataframes)
+
+    pivot_df = result.pivot(index="info", columns="name", values=parameter)
+
+    colors = []
+    # For this data I think tab20 was the best color choice
```

Comment:
I'd print a warning if > 20 functions are plotted. I mean it would be a stupid thing to do anyway, but it would probably confuse the user that two adjacent bars are identically colored.